File: PDOO/Material Complementario/Tema7/VillanosRB-ClaseTeoria/lib/villano.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Villanos
  class Villano
    @@Asociacion = "VillanosSA"

    def initiliaze(n,e,f)
      @nombre = n
      @escondite_secreto = e
      @fortuna = f
      @planes = Array.new
      @mayordomo = nil
    end

    def set_fortuna(cantidad)

    end

    def seguir_plan_prioritario()

    end

    def primer_plan_activo()

    end

    def self.set_asociacion(asociacion)

    end
    private :set_fortuna

    # si el set_asociacion fuera privado
    # private_class_method :set_asociacion
  end
end

File: PDOO/Material Complementario/Tema7/Villanos-ClaseTeoria/src/villanos/Villano.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package villanos;

import java.util.ArrayList;
import esbirros.Mayordomo;

/**
 *
 * @author salva
 */
public class Villano {

    //atributos de instancia
    public String nombre;
    String esconditeSecreto;
    private int fortuna;

    //atributos de clase
    public static String Asociacion = "VillanosSA";

    //atributos de referencia
    private ArrayList<Plan> planes;
    Mayordomo mayordomo;

    //constructor por defecto
    public Villano(String nombre, String esconditeSecreto, int fortuna) {
        this.nombre = nombre;
        this.esconditeSecreto = esconditeSecreto;
        this.fortuna = fortuna;
        this.mayordomo = null;
        this.planes = new ArrayList<>();
        planes.add(new Plan());
    }

    private void setFortuna(int cantidad) {

    }

    public boolean seguirPlanPrioritario() {

    }

    public Plan primerPlanActivo() {

    }

    public static void setAsociacion(Asociacion Asociacion) {

    }
}
